src/pulp/solvers.py

```diff
@@ -1420,7 +1420,11 @@ def solve_CBC(self, lp, use_mps=True):
         args.append(self.path)
         args.extend(cmds[1:].split())
         cbc = subprocess.Popen(args, stdout = pipe, stderr = pipe)
-        if cbc.wait() != 0:
+
+        if processTimeout is not None and sys.version_info < (3, 3):
+            warnings.warn('Pulp: Subprocess timeout is not available in python < 3.3. Solver will run indefinitely.')
+
+        if (cbc.wait(processTimeout) if sys.version_info >= (3, 3) else cbc.wait()) != 0:
```

Comment:
As `cbc.wait(timeout)` can raise a `TimeoutExpired` exception, one could consider to catch this exception and raise a `PulpSolverError` instead. But I am not sure which kind of interface is desired here in the project.